src/emqx_ct_helpers_docker.erl

```diff
@@ -31,24 +31,65 @@ inspect( ContainerName, Inspection ) ->
     docker("inspect -f "++Inspection++" "++ContainerName).
 
 docker(Cmd) ->
-    os:cmd("docker "++Cmd).
+    sh("docker "++Cmd).
 
 compose( File, Project_name, ServiceName, Args, Env ) ->
     { ok, EnvPath } = set_env_file(Env),
     Cmd = "docker-compose --project-name "++Project_name++" --env-file "++EnvPath++" "++Args
           ++" -f "++File++" up --detach "++ServiceName,
-    Res = os:cmd(Cmd),
-    Res.
+    sh(Cmd).
 
 stop(ContainerName) ->
     Cmd = "docker stop " ++ ContainerName,
-    Res = os:cmd(Cmd),
-    Res.
+    sh(Cmd).
 
 set_env_file(Env) ->
     { ok, CWD } = file:get_cwd(),
     Unique = erlang:unique_integer()*-1,
     Filename = lists:concat([ ".docker-env-", Unique ]),
     Path = filename:join([ CWD, Filename]),
-    file:write_file(Path, Env),
+    ok = file:write_file(Path, Env),
     { ok, Path }.
+
```

Comment:
have you considered this before, the last time I wanted to better exit status it used :

```erlang
os_cmd(Command) ->
	Port = open_port({spawn, Command}, [ use_stdio, stderr_to_stdout, eof, hide, exit_status, { parallelism, true } ]),
	get_data(Port, []).

get_data(Port, Sofar) ->
	receive
		{Port, {data, Bytes}} ->
			get_data(Port, [Sofar|Bytes]);
		{Port, eof} ->
			Port ! {self(), close},
			receive
				{ Port, closed } ->
					true
			end,
			receive
				{'EXIT',  Port,  _} ->
					ok
			after 1 ->              % force context switch
				ok
			end,
			ExitCode = receive
						 { Port, { exit_status, Code } } ->
							Code
					   end,
			{ ExitCode, lists:flatten( Sofar) }
	end.
```

Reply:
Yes. I used both before.
but I find `echo $?` is easier to understand.